src/libs/Navigation/AppNavigator/AuthScreens.tsx

```diff
@@ -35,24 +32,51 @@ import * as SessionUtils from '../../SessionUtils';
 import NotFoundPage from '../../../pages/ErrorPage/NotFoundPage';
 import getRootNavigatorScreenOptions from './getRootNavigatorScreenOptions';
 import DemoSetupPage from '../../../pages/DemoSetupPage';
+import * as OnyxTypes from '../../../types/onyx';
+import type {WindowDimensions} from '../../../styles/getModalStyles';
+import type {AuthScreensStackParamList} from './types';
 
-let timezone;
-let currentAccountID;
-let isLoadingApp;
+type AuthScreensOnyxProps = {
+    /** Session of currently logged in user */
+    session: OnyxEntry<OnyxTypes.Session>;
+
+    /** The report ID of the last opened public room as anonymous user */
+    lastOpenedPublicRoomID: OnyxEntry<string>;
```

Comment:
I think its not a OnyxEntry so I would change it to string | null or I think in type-fest there is type for this, also I think it can be undefined for example if we didnt open any public room earlier

Reply:
@kubabutkiewicz If I make it optional, there are some type errors in withOnyx HOC
![image](https://github.com/Expensify/App/assets/23176449/3643ba86-3733-4475-a7f2-0fcb8bbc1241)